slide_analysis/UI/image_viewer.py

```diff
@@ -3,110 +3,110 @@
 # from PyQt5.QtWidgets import (QAction, QApplication, QFileDialog, QLabel,
 #                              QMainWindow, QMenu, QMessageBox, QScrollArea, QSizePolicy)
 #
+from PyQt5 import QtGui, QtCore
+
 from PyQt5.QtCore import QDir, Qt
 from PyQt5.QtGui import QPalette, QPixmap
-from PyQt5.QtWidgets import QMainWindow, QFileDialog, QMessageBox, QAction, QApplication, QMenu
+from PyQt5.QtWidgets import *
 
 from slide_analysis.UI.image_helper import ImageHelper
 from slide_analysis.UI.ui_mainWindow import Ui_MainWindow
 
 
+class MyPopup(QWidget):
+    def __init__(self, tile):
+        QWidget.__init__(self)
+        self.tile_label = QLabel(self)
+        pixmap = QPixmap.fromImage(tile)
+        self.tile_label.setPixmap(pixmap)
+        self.resize(pixmap.width(), pixmap.height())
+
+    def keyPressEvent(self, event):
+        if event.key() == QtCore.Qt.Key_Space:
+            self.close()
+            event.accept()
+        else:
+            super().keyPressEvent(event)
+
+
 class ImageViewer(QMainWindow, Ui_MainWindow):
     def __init__(self):
         super(ImageViewer, self).__init__()
+        self.image_helper = None
         self.setupUi(self)
         self.show()
 
         self.scale_factor = 0.0
 
-        # self.imageLabel = QLabel()
         self.imageLabel.setBackgroundRole(QPalette.Base)
         # self.imageLabel.setSizePolicy(QSizePolicy.Ignored, QSizePolicy.Ignored)
         # self.imageLabel.setScaledContents(True)
 
-        # self.scrollArea = QScrollArea()
-        # self.scrollArea.setBackgroundRole(QPalette.Dark)
-        # self.scrollArea.setWidget(self.imageLabel)
-        # self.setCentralWidget(self.scrollArea)
-
         self.create_actions()
         self.create_menus()
 
-        self.setWindowTitle("Image Viewer")
-        # self.resize(500, 400)
+    def resizeEvent(self, event):
+        if self.image_helper is not None:
```

Comment:
move all  `if self.image_helper is not None:`
to method to avoid code duplicating